classes/order/OrderCartRule.php

```diff
@@ -62,7 +62,7 @@ class OrderCartRuleCore extends ObjectModel
             'id_order' => ['type' => self::TYPE_INT, 'validate' => 'isUnsignedId', 'required' => true],
             'id_cart_rule' => ['type' => self::TYPE_INT, 'validate' => 'isUnsignedId', 'required' => true],
             'id_order_invoice' => ['type' => self::TYPE_INT, 'validate' => 'isUnsignedId'],
-            'name' => ['type' => self::TYPE_STRING, 'validate' => 'isCleanHtml', 'required' => true],
+            'name' => ['type' => self::TYPE_HTML, 'required' => true],
```

Comment:
```suggestion
            'name' => ['type' => self::TYPE_STRING, 'required' => true],
```
Same, it should be a string, but without any validation, and accept all characters. We don't expect to receive HTML inside this one

Reply:
No. With `TYPE_STRING`, the value is strip_tagged. So all characters are not allowed.